lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,165 @@
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { GithubHttp } from '../../../util/http/github';
+import { regEx } from '../../../util/regex';
+import { streamToString } from '../../../util/streams';
+import { parseUrl } from '../../../util/url';
+import { id } from '../../versioning/hermit';
+import { Datasource } from '../datasource';
+import { getApiBaseUrl } from '../github-releases/common';
+import type {
+  GithubRelease,
+  GithubReleaseAsset,
+} from '../github-releases/types';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { HermitSearchResult } from './types';
+
+/**
+ * Hermit Datasource searches a given package from the specified `hermit-packages`
+ * repository. It expects the search manifest to come from an asset `index.json` from
+ * a release named index.
+ */
+export class HermitDatasource extends Datasource {
+  static readonly id = 'hermit';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = id;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/cashapp/hermit-packages',
+  ];
+
+  pathRegex: RegExp;
+
+  constructor() {
+    super(HermitDatasource.id);
+    this.http = new GithubHttp(id);
+    this.pathRegex = regEx('^\\/(?<owner>[^/]+)\\/(?<repo>[^/]+)$');
+  }
+
+  @cache({
+    namespace: `datasource-hermit-package`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? ''}-${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    logger.trace(`HermitDataSource.getReleases()`);
+
+    if (!registryUrl) {
+      logger.error('registryUrl must be supplied');
+      return null;
+    }
+
+    const parsedUrl = parseUrl(registryUrl);
+    if (parsedUrl === null) {
+      logger.warn({ registryUrl }, 'invalid registryUrl given');
+      return null;
+    }
+
+    if (!registryUrl.startsWith('https://github.com/')) {
+      logger.warn({ registryUrl }, 'Only Github registryUrl is supported');
+      return null;
+    }
+
+    const items = await this.getHermitSearchManifest({
+      packageName,
+      registryUrl,
```

Comment:
pass `parsedUrl`, so you don't need to parse again

Reply:
I did pass it in a previous commit. however yesterday when I was trying it. the `key` function for cache in `getHermitSearchManifest` gets an empty object instead. where I couldn't turn it into a string like I do now.  It causes error when I run Renovate against our environment. 

Not sure it Is a known issue with the `cache` decorator.?

Reply:
tried it again. the empty object seems to relate to the logger filtering. key function now properly returns string of the given parsed url object.  might did something wrong yesterday. never mind